src/main/java/com/notably/model/BlockModel.java

```diff
@@ -0,0 +1,53 @@
+package com.notably.model;
+
+import com.notably.commons.core.path.AbsolutePath;
+import com.notably.model.block.Block;
+import com.notably.model.block.BlockTree;
+import com.notably.model.block.BlockTreeItem;
+
+/**
+ * API of the BlockModel component.
+ */
+public interface BlockModel {
+    /**
+     * Gets the BlockTree currently in use by Notably.
+     */
+    BlockTree getBlockTree();
+
+    /**
+     * Check if currently open block has the specified child block.
+     */
+    boolean hasChild(Block b);
+
+    /**
+     * Adds a block to the specified path, if possible.
+     */
+    void addBlock(AbsolutePath p, Block b);
+
+    /**
+     * Removes the block at the specified path, if possible.
+     */
+    void removeBlock(AbsolutePath p);
+
+    /**
+     * Opens the block at the specified path, if possible.
+     */
+    BlockTreeItem openBlock(AbsolutePath p);
+
+    /**
+     * Sets the currently open block to a new block. (Replaces its block content)
+     */
+    void saveBlock(Block newBlock);
```

Comment:
Is this for the `edit` command?
If it is, should we change this to `updateCurrentlyOpenBlock` or `editCurrentlyOpenBlock` or something else?
Also, shouldn't we restrict the parameter to only the new body? (since we're not allowing title edits for the moment)

Reply:
> Is this for the `edit` command?

Yes!
> Also, shouldn't we restrict the parameter to only the new body? (since we're not allowing title edits for the moment)

Hmm... I guess that true. If that's the case then, `updateCurrentlyOpenBlockBody` would be a better name IMO

Reply:
Okay!